src/types/hooks/auth/use-auth.ts

```diff
@@ -1,28 +1,32 @@
 import { useCallback, useMemo } from "react";
 import { useSelector } from "react-redux";
-import { useLogoutMutation } from "../../../api/auth";
+import { useLogoutMutation, useRefreshAccesTokenMutation } from "../../../api/auth";
+import { StatusCodes } from "../../../constants/api/status-codes";
 import { useAppDispatch } from "../../../hooks/store/use-app-dispatch-hook";
 import { useAppSelector } from "../../../hooks/store/use-app-selector-hook";
-import { removeCredentials, selectIsAuthenticated, user } from "../../../store/slices/auth-slice";
+import { removeCredentials, selectIsAuthenticated, setAccesToken, user } from "../../../store/slices/auth-slice";
 
-
-const useAuth = () => {
+export const useAuth = () => {
   const dispatch = useAppDispatch();
   const authUser = useAppSelector(user);
   const isAuthorized = useSelector(selectIsAuthenticated);
   const [logout, { isLoading: isLogoutLoading, isError: isLogoutError}] = useLogoutMutation();
+  const [newRefreshToken, {isLoading: isRefreshLoading, isError: isRefreshError}] = useRefreshAccesTokenMutation();
   const logOut = useCallback(async () => {
     if (!isAuthorized) {
       try {
-        await logout(null); ////refreshtoken in swager
-        dispatch(removeCredentials());
+        const response = await newRefreshToken(null).unwrap();
+        if (response.status === StatusCodes.UNAUHTORIZED) {///error ?
+          dispatch(removeCredentials());
+        } else {
+          dispatch(setAccesToken(response.accessToken));
+
+          //// make get req 
+        }
```

Comment:
there you just need to send request for removing refresh from cookies and independ of request status remove access from local storage (just use it 'dispatch(removeCredentials());')